Backend/src/main/java/com/abhinternship/auctionapp/controller/UserController.java

```diff
@@ -0,0 +1,30 @@
+package com.abhinternship.auctionapp.controller;
+
+import com.abhinternship.auctionapp.model.User;
+import com.abhinternship.auctionapp.repository.UserRepository;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
+public class UserController {
+    private UserRepository userRepository;
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    public UserController(UserRepository userRepository,
+                          BCryptPasswordEncoder bCryptPasswordEncoder) {
+        this.userRepository = userRepository;
+        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
+    }
+
+    @PostMapping("/authentication/register")
+    public void signUp(@RequestBody User user) {
+        System.out.println(user.getFirstName());
+        System.out.println(user.getEmail());
+        user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
```

Comment:
Create a base controller with get, create, delete and update methods, and rename this to create and override from base controller. Base controller will call baseService.create(...), that you also need to make. 
Call userService.create from this overriden method, and from there hash the password and and call repository save method.